### ISSUE TITLE
unit tests are failing when run with java 11

### ISSUE THREAD

**Issue by adietish** [openshift/openshift-restclient-java#398]
this started happening when #394 - which upgraded okHttp to 4 - was merged.

**Comment by adietish** [comment]
travis output with java 11:
```
Tests in error: 
  testDeserialization(com.openshift.internal.restclient.model.v1.EndpointsTest): java.lang.reflect.InvocationTargetException-->null
  testBuildingURLForAWatchService(com.openshift.internal.restclient.URLBuilderTest): java.lang.reflect.InvocationTargetException-->null
  testDuplicateParameters(com.openshift.internal.restclient.URLBuilderTest): java.lang.reflect.InvocationTargetException-->null
  testBuildingURLForAProjectUsingResource(com.openshift.internal.restclient.URLBuilderTest): java.lang.reflect.InvocationTargetException-->null
  testBaseURLWithTrailingSlash(com.openshift.internal.restclient.URLBuilderTest): java.lang.reflect.InvocationTargetException-->null
  testAddingASubResource(com.openshift.internal.restclient.URLBuilderTest): java.lang.reflect.InvocationTargetException-->null
  testAddingASubContext(com.openshift.internal.restclient.URLBuilderTest): java.lang.reflect.InvocationTargetException-->null
  testKubernetesMasterVersionOk(com.openshift.internal.restclient.DefaultClientConstructionTest): java.lang.reflect.InvocationTargetException-->null
  testKubernetesMasterVersion404(com.openshift.internal.restclient.DefaultClientConstructionTest): java.lang.reflect.InvocationTargetException-->null
  testOpenShiftMasterVersionOk(com.openshift.internal.restclient.DefaultClientConstructionTest): java.lang.reflect.InvocationTargetException-->null
  testOpenShiftMasterVersion404(com.openshift.internal.restclient.DefaultClientConstructionTest): java.lang.reflect.InvocationTargetException-->null
  testKubernetesResourceIsSupportedAfterInitiallyErrorIsThrown(com.openshift.internal.restclient.ApiTypeMapperTest): java.lang.reflect.InvocationTargetException-->null
  testKubernetesResourceIsSupported(com.openshift.internal.restclient.ApiTypeMapperTest): java.lang.reflect.InvocationTargetException-->null
  testOpenShiftResourceIsSupported(com.openshift.internal.restclient.ApiTypeMapperTest): java.lang.reflect.InvocationTargetException-->null
  testRandomResourceIsNotSupported(com.openshift.internal.restclient.ApiTypeMapperTest): java.lang.reflect.InvocationTargetException-->null
  testApiGroupResourceIsSupported(com.openshift.internal.restclient.ApiTypeMapperTest): java.lang.reflect.InvocationTargetException-->null
  testIsSupported(com.openshift.internal.restclient.api.capabilities.PodExecTest): java.lang.reflect.InvocationTargetException-->null
  testIsNotSupportedWhenEndpointDoesNotExist(com.openshift.internal.restclient.api.capabilities.PodExecTest): java.lang.reflect.InvocationTargetException-->null
  testAdapterCallsListenerCycle(com.openshift.internal.restclient.api.capabilities.PodExecTest): java.lang.reflect.InvocationTargetException-->null
  testExecOptions(com.openshift.internal.restclient.api.capabilities.PodExecTest): java.lang.reflect.InvocationTargetException-->null
  testIsSupported(com.openshift.internal.restclient.capability.resources.PodLogRetrievalAsyncTest): java.lang.reflect.InvocationTargetException-->null
  testIsNotSupportedWhenEndpointDoesNotExist(com.openshift.internal.restclient.capability.resources.PodLogRetrievalAsyncTest): java.lang.reflect.InvocationTargetException-->null
  testAdapterCallsListenerCycle(com.openshift.internal.restclient.capability.resources.PodLogRetrievalAsyncTest): java.lang.reflect.InvocationTargetException-->null
  testStopWhenConnected(com.openshift.internal.restclient.capability.resources.PodLogRetrievalAsyncTest): java.lang.reflect.InvocationTargetException-->null
  testStopSwallowsException(com.openshift.internal.restclient.capability.resources.PodLogRetrievalAsyncTest): java.lang.reflect.InvocationTargetException-->null
  client_should_equal_client_with_same_user_with_different_token(com.openshift.internal.restclient.DefaultClientTest): java.lang.reflect.InvocationTargetException-->null
  clientShouldEqualClientWithSameUrl(com.openshift.internal.restclient.DefaultClientTest): java.lang.reflect.InvocationTargetException-->null
  clientShouldNotEqualClientWithDifferentUrl(com.openshift.internal.restclient.DefaultClientTest): java.lang.reflect.InvocationTargetException-->null
  client_should_not_equal_client_with_different_username(com.openshift.internal.restclient.DefaultClientTest): java.lang.reflect.InvocationTargetException-->null
  should_use_paylod_in_delete_request(com.openshift.internal.restclient.DefaultClientTest): java.lang.reflect.InvocationTargetException-->null
  should_use_paylod_in_post_request(com.openshift.internal.restclient.DefaultClientTest): java.lang.reflect.InvocationTargetException-->null
  should_use_paylod_in_put_request(com.openshift.internal.restclient.DefaultClientTest): java.lang.reflect.InvocationTargetException-->null
  should_not_use_paylod_in_get_request(com.openshift.internal.restclient.DefaultClientTest): java.lang.reflect.InvocationTargetException-->null
  should_not_use_paylod_in_head_request(com.openshift.internal.restclient.DefaultClientTest): java.lang.reflect.InvocationTargetException-->null
  should_not_send_resource_payload_when_deleting(com.openshift.internal.restclient.DefaultClientTest): java.lang.reflect.InvocationTargetException-->null

Tests run: 421, Failures: 0, Errors: 35, Skipped: 0
```

**Comment by adietish** [comment]
fixed in #399
